crates/bitwarden-vault/src/cipher/cipher_client/edit.rs

```diff
@@ -0,0 +1,718 @@
+            id: value.id.ok_or(MissingFieldError("id"))?,
+            r#type: value.r#type,
+            organization_id: value.organization_id,
+            folder_id: value.folder_id,
+            favorite: value.favorite,
+            reprompt: value.reprompt,
+            key: value.key,
+            name: value.name,
+            notes: value.notes,
+            fields: value.fields.unwrap_or_default(),
+            password_history: value.password_history,
+            attachments: value.attachments,
+            type_data,
+            revision_date: value.revision_date,
+            archived_date: value.archived_date,
+        })
+    }
+}
+
+impl CipherEditRequest {
+    fn update_password_history(&mut self, original_cipher: &CipherView) {
+        let changes = self
+            .detect_login_password_changes(original_cipher)
+            .into_iter()
+            .chain(self.detect_hidden_field_changes(original_cipher));
+        let history: Vec<_> = changes
+            .rev()
+            .map(|change| change.into_history_entry())
+            .chain(original_cipher.password_history.iter().flatten().cloned())
+            .take(MAX_PASSWORD_HISTORY_ENTRIES)
+            .collect();
+
+        self.password_history = (!history.is_empty()).then_some(history);
+    }
+
+    fn detect_login_password_changes(
+        &mut self,
+        original_cipher: &CipherView,
+    ) -> Vec<PasswordChange> {
+        if self.r#type != CipherType::Login || original_cipher.r#type != CipherType::Login {
+            return Default::default();
+        }
+
+        let (Some(original_login), Some(current_login)) = (
+            original_cipher.login.as_ref(),
+            self.type_data.as_login_view_mut(),
+        ) else {
+            return Default::default();
+        };
+
+        let original_password = original_login.password.as_deref().unwrap_or("");
+        let current_password = current_login.password.as_deref().unwrap_or("");
+
+        if original_password.is_empty() {
+            // No original password - set revision date only if adding new password
+            if !current_password.is_empty() {
+                current_login.password_revision_date = Some(Utc::now());
+            }
+            Default::default()
+        } else if original_password == current_password {
+            // Password unchanged - preserve original revision date
+            current_login.password_revision_date = original_login.password_revision_date;
+            Default::default()
+        } else {
+            // Password changed - update revision date and track change
+            current_login.password_revision_date = Some(Utc::now());
+            vec![PasswordChange::new_password(original_password)]
+        }
+    }
+
+    fn detect_hidden_field_changes(&self, original_cipher: &CipherView) -> Vec<PasswordChange> {
+        let original_fields =
+            Self::extract_hidden_fields(original_cipher.fields.as_ref().unwrap_or(&vec![]));
```

Comment:
With the change to accept a slice you can update this to use default.
```suggestion
            Self::extract_hidden_fields(original_cipher.fields.as_ref().unwrap_or_default();
```